libvcx/src/settings.rs

```diff
@@ -80,6 +83,7 @@ pub static MOCK_DEFAULT_INDY_PROOF_VALIDATION: &str = "true";
 
 lazy_static! {
     static ref SETTINGS: RwLock<HashMap<String, String>> = RwLock::new(HashMap::new());
+    pub static ref AGENCY_CLIENT: Mutex<AgencyClient> = Mutex::new(AgencyClient::default()); // TODO: Do need a mutex to make it mutable?
```

Comment:
In general yes, since this is globally visible across threads, but in this case RwLock fits better. Unlike Mutex, it allows for multiple readers.

Reply:
Why not rwlock?